src/metabase/pulse.clj

```diff
@@ -419,10 +419,57 @@
+  :on-change reconfigure-retrying
+  :deprecated "0.44.1"
+  :setter (fn notification-retry-randomizaion-factor-setter [new-value]
+            (when (some? new-value)
+              (throw (ex-info (trs (str "The setting :notification-retry-randomizaion-factor is deprecated, "
+                                        "please use :notification-retry-randomization-factor instead."))
+                              {})))
+            (setting/set-value-of-type! :double :notification-retry-randomizaion-factor nil)))
+
+(defn- notification-retry-randomization-factor-misspelled-env? []
+  (some? (setting/env-var-value :notification-retry-randomizaion-factor)))
+
+(defn- resolve-misspelled-randomization-factor!
+  "Returns the value of the :notification-retry-randomization-factor setting migrating
+  the mispelled variant :notification-retry-randomizaion-factor if possible. If the deprecated
+  setting is in use via the environment variable, it issues a warning and uses that value.
+
+  When we can get rid of the deprecated :notification-retry-randomization-factor setting,
+  we have to set the :default value (0.1) for :notification-retry-randomization-factor and
+  remove its :getter and :setter as well as this function."
+  []
+  (if-some [correct-spelling-value (setting/get-value-of-type :double :notification-retry-randomization-factor)]
+    correct-spelling-value
+    (if-some [misspelled-value (notification-retry-randomizaion-factor)]
+      (do
+        (if (notification-retry-randomization-factor-misspelled-env?)
+          (log/warn
+           (trs (str "The setting MB_NOTIFICATION_RETRY_RANDOMIZAION_FACTOR is deprecated, "
+                     "please set MB_NOTIFICATION_RETRY_RANDOMIZATION_FACTOR instead.")))
+          (do
+            (log/warn
+             (trs (str "Migrating deprecated setting notification-retry-randomizaion-factor "
+                       "to notification-retry-randomization-factor")))
+            (setting/set-value-of-type! :double :notification-retry-randomization-factor misspelled-value)
+            (notification-retry-randomizaion-factor! nil)))
+        misspelled-value)
+      0.1)))
+
+(defsetting notification-retry-randomization-factor
   (deferred-tru "The randomization factor of the retry delay when delivering notifications.")
   :type :double
-  :default 0.1
-  :on-change reconfigure-retrying)
+  :on-change reconfigure-retrying
+  :getter resolve-misspelled-randomization-factor!
+  :setter (fn notification-retry-randomization-factor-setter [new-value]
+            (when (notification-retry-randomization-factor-misspelled-env?)
+              (throw (ex-info (trs (str "The environment variable MB_NOTIFICATION_RETRY_RANDOMIZAION_FACTOR is set, "
+                                        "unset it to be able to set :notification-retry-randomization-factor."))
+                              {})))
+            (setting/set-value-of-type! :double :notification-retry-randomization-factor new-value)))
```

Comment:
@dpsutton I've implemented this check, but I'm not 100% sure it's a clear improvement.
It can potentially make it one step more difficult to configure the setting by requiring that the user clean up the configuration, but there are other ways for getting unclean setups that we don't check. (E.g., configuring both environment variables or both database settings.)
I could imagine both removing this check as well as emitting a warning whenever the new setting is used while the old one is still configured. WDYT?

Reply:
@flamber mentioned the idea that it might be fine to just correct this typo and carry on without trying to backfill it. Maybe we just patch it up and don't fret since it is so new?

Reply:
I doubt anybody uses this parameter. What would the patch you suggest look like? Just inserting a `t`?

Reply:
Yeah. The idea was to just correct the misspelling and carry on. Seems reasonable to me.